contrib/pg_stat_statements/sql/test_pg_stat_statements.sql

```diff
@@ -0,0 +1,215 @@
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+
+-- start tests for pg_stat_statements logic
+
+-- Known issue: query is not added to pg_stat_statements statistics in
+-- case it is planned by GPORCA. So disable GPORCA during tests.
+SET optimizer='off';
+
+SELECT pg_stat_statements_reset();
+
+-- launch 2 equivalent queries  
+SELECT
+    GROUPING(item1) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+
+SELECT
+    GROUPING(item1) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 1
+GROUP BY ROLLUP(item1, item2);
+
+-- pg_stat_statements statistics should have 2 calls for 1 entry
+-- corresponding to the two preceding SELECT queries
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- launch not equivalent query
+SELECT
+    GROUPING(item2) AS item_1_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+
+-- check that it has separate entry
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- change alias name
+SELECT
+    GROUPING(item2) AS item_2_grouping
+FROM
+    table_test_pg_stat_statements
+WHERE
+    item3 = 0
+GROUP BY ROLLUP(item1, item2);
+-- check that recent request was added to preceding entry
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check that different grouping options result in separate entries
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item1, item2, item3, item4);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item1, item2, item3, item4);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY GROUPING SETS(item1, item2, item3, item4);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY GROUPING SETS((item1, item2), (item3, item4));
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY item1, item2, item3, item4;
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several parameters options in ROLLUP
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item1, item2);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item2, item3);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY ROLLUP(item3, item4);
+
+SELECT query, calls FROM pg_stat_statements ORDER BY QUERY;
+
+-- check several parameters options in CUBE
+-- all should result in separate entries
+SELECT pg_stat_statements_reset();
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item1, item2);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item2, item3);
+
+SELECT
+    COUNT (*)
+FROM
+    table_test_pg_stat_statements
+GROUP BY CUBE(item3, item4);
```

Comment:
the logic of attribute processing for ROLLUP and CUBE is the same, I think it's worth leaving one thing (is there anything specific in processing GROUPING SETS, if not, then they should also be removed). I also think two queries will be enough for the test, not three.

Reply:
Ok, will leave 1 option with 2 queries